### PR TITLE
Upgrade to Vue3

### PR DESCRIPTION
**Pull request by AlecTroemel**
Also includes a new `debounce` prop which debounces the emitted `update:value` event

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,81 +1,68 @@
...
-        "> 1%",
-        "last 2 versions",
-        "ie >= 9"
-    ],
-    "dependencies": {
-        "vite-plugin-css-injected-by-js": "^3.3.0",
-        "vue-template-compiler": "^2.7.14"
-    }
+	"name": "@mirus/tiptap-editor",
+	"version": "2.1.9",
```
Comment (Swinejamin) [inline package.json:3]: @AlecTroemel we need to bump the version here, right?

File: package.json [hunk 2]
```diff
@@ -1,81 +1,68 @@
...
+		"@tiptap/extension-history": "^2.0.3",
+		"@tiptap/extension-italic": "2.4.0",
+		"@tiptap/extension-list-item": "2.4.0",
+		"@tiptap/extension-paragraph": "2.4.0",
+		"@tiptap/extension-placeholder": "2.4.0",
+		"@tiptap/extension-text": "2.4.0",
+		"@tiptap/pm": "2.4.0",
+		"@tiptap/vue-3": "2.4.0",
+		"@vitest/coverage-v8": "^0.32.2",
+		"@vue/compat": "^3.1.0",
```
Comment (Swinejamin) [inline package.json:49]: Is the compat build still needed?

File: package.json (hunk 1 above)
Comment (AlecTroemel) [reply package.json:3]: ah yep definitely do!

File: package.json (hunk 2 above)
Comment (AlecTroemel) [reply package.json:49]: good question. I dont think so, but I'll double check

Comment (AlecTroemel) [reply package.json]: ya @vue/compat is specifically for vue2. Away it goes!

**Review by Swinejamin — APPROVED** [review]

**Pull request merged**